test/e2e/accounts/snap-account-contract-interaction.spec.ts

```diff
@@ -0,0 +1,64 @@
+import GanacheContractAddressRegistry from '../seeder/ganache-contract-address-registry';
+import { scrollAndConfirmAndAssertConfirm } from '../tests/confirmations/helpers';
+import {
+  createDepositTransaction,
+  TestSuiteArguments,
+} from '../tests/confirmations/transactions/shared';
+import {
+  multipleGanacheOptionsForType2Transactions,
+  withFixtures,
+  openDapp,
+  WINDOW_TITLES,
+} from '../helpers';
+import FixtureBuilder from '../fixture-builder';
+import { SMART_CONTRACTS } from '../seeder/smart-contracts';
+import { installSnapSimpleKeyring, importKeyAndSwitch } from './common';
+
+describe('Snap Account - Contract interaction', function () {
+  const smartContract = SMART_CONTRACTS.PIGGYBANK;
+
+  it('deposits to piggybank contract', async function () {
```

Comment:
Lets expand this test to also check that the balance is correct after the contract interaction

Reply:
added, thanks!